### DISCUSSION TITLE
Shared stuff and ActivityPub

### DISCUSSION THREAD

**Discussion by macram** [FreshRSS/FreshRSS#5219] [Ideas]
Hello everyone!

I'm a new FreshRSS user. I'm amazed, the software is great, nice UI and nice workings. Congratulations to all contributors!

I was fantasizing with sharing articles. I haven't found a roadmap or something similar and I'm pretty sure you have already considered this. First, it would be great to be able to share articles with another users of the same instance (I'm using this word on purpose), a la Google Reader.

And it would be even amazing to have some kind of compatibility with the Fediverse using ActivityPub (I know this is the current hype but I had to propose this before I forget the idea). The ability to follow users on another FreshRSS instances and even Mastodon/Pleroma accounts who posts links, and the ability to be followed from there. 

I know this is not exactly core for a software like this but I think it would be amazing, and I had to share it and to propose it. I'm afraid I'm not exactly proficient on PHP so I can't code it myself.

Thank you very much!

**Comment by Alkarex** [comment]
Hello,
A first step would be to improve our existing sharing via RSS/Atom, and then also via HTML.
The main core feature to improve for that to be possible is a new approach to our tokens. See https://github.com/FreshRSS/FreshRSS/issues/3178
Indeed, we already have the ability to republish any FreshRSS view as RSS, but at the moment it is the same token for everything, which is not intended for public use.

We also have the ability to manually share single articles from within FreshRSS to multiple platforms, including Mastodon.

For ActivityPub, which is a big chunk, an initial approach can be to use an RSS to ActivityPub bridge.

**Comment by macram** [comment]
Hello,

If I'm understanding well, that type of feeds could be great grounds to develop this feature.

As I'm imagining it, it would need the following elements:

- UI element which allows to _share_, perhaps this would mean assigning a tag to an article. Perhaps add the ability to add a comment when sharing. 
- The ability for each user to have a "public" feed with a distinct token (or no token at all) to syndicate these articles.
- These feeds should show slightly different than a usual RSS feed. Why? Because in the future some features could be built around them (comments, resharing...).
- A RSS to ActivityPub bridge allowing to follow this from fediverse, and perhaps easing the way to find content from another FreshRSS instance. 

Some questions arise:
- What would happen when replying to a FreshRSS shared item from Mastodon? It should be showed on FreshRSS UI IMHO, allowing to reply, but this is a whole another feature. 
- Perhaps the whole social thingie could be leveraged to an ActivityPub integration? This way a FreshRSS user could get links shared by a Mastodon user. Also replies could be handled this way. But I'm not sure how much work is this (I'm inclined to think that's too much) and perhaps we can find a way to keep it simpler.

**Reply by Alkarex** [reply]
The approach I have in mind at the moment is to use our existing tagging system to chose what to share and where (that almost works already, minus the improved token stuff I mentioned). Potentially supplemented by a new way to attach a text note to an article. But not much in the other direction yet for responses (except by subscribing to the corresponding RSS feed). For more sophisticated use cases, sharing with a tool more focussed on that is already possible, e.g. Mastodon, Friendica, Wallabag, Shaarli, etc.

**Reply by macram** [reply]
That's great, only with a dedicated button to "Recommend" the current post with the note, not leaving the user to write the tag by themself.

**Reply by macram** [reply]
Only FYI and regarding this, I just learned Mastodon has RSS feeds. We only need to append .rss to the URL. So with these "public curated RSS" and perhaps a RSS-to-ActivityPub bridge this would be good to go.

**Reply by Frenzie** [reply]
Cross-ref to https://github.com/FreshRSS/FreshRSS/pull/5079#issuecomment-1421565933

**Comment by Alkarex** [comment]
For reference: https://rss-bridge.github.io/rss-bridge/Bridge_Specific/ActivityPub_(Mastodon).html
